master/buildbot/test/unit/test_status_github.py

```diff
@@ -170,11 +170,28 @@ def test_startService(self):
 
         self.status._status.subscribe.assert_called_with(self.status)
 
+    def test_stopService(self):
+        """
+        When stopped it will unsubscribe from all statuses.
+        """
+        builderstatus = FakeBuilderStatus()
+
+        result = self.status.builderAdded('builder-name', builderstatus)
```

Comment:
The code in 'master/buildbot/status/builder.py' is just dead then? Our production and test environments are stuck on buildbot eight. Unfortunately there aren't any features that are compelling enough to push us to nine, and we've added a bunch of features on top of eight that don't port to nine, so moving isn't going to happen any time soon.